politeiad/cmd/legacypoliteia/import.go

```diff
@@ -0,0 +1,335 @@
+func importRecord(p *proposal, tstoreToken []byte, cmd *importCmd) error {
+	// Prepare vote metadata file if needed.
+	if p.VoteMetadata.LinkBy != 0 || p.VoteMetadata.LinkTo != "" {
+		b, err := json.Marshal(p.VoteMetadata)
+		if err != nil {
+			return err
+		}
+		p.Files = append(p.Files, backend.File{
+			Name:    "votemetadata.json",
+			MIME:    mime.DetectMimeType(b),
+			Digest:  hex.EncodeToString(util.Digest(b)),
+			Payload: base64.StdEncoding.EncodeToString(b),
+		})
+	}
+
+	// Prepare status changes metadata stream.
+	for _, scmd := range p.StatusChanges {
+		b, err := json.Marshal(scmd)
+		if err != nil {
+			return err
+		}
+		p.Metadata = append(p.Metadata, backend.MetadataStream{
+			PluginID: usermd.PluginID,
+			StreamID: usermd.StreamIDStatusChanges,
+			Payload:  string(b),
+		})
+	}
+
+	// Check if record status is public. If so, we need to first save it as
+	// unreviewed, and then save it as public. This is done to bypass the
+	// validations from the RecordSave function.
+	isPublic := false
+	if p.RecordMetadata.Status == backend.StatusPublic {
+		isPublic = true
+		p.RecordMetadata.Status = backend.StatusUnreviewed
+	}
+
+	// Save record to tstore.
+	err := cmd.tstore.RecordSave(tstoreToken, p.RecordMetadata, p.Metadata,
+		p.Files)
+	if err != nil {
+		return err
+	}
+
+	// If public, update from unreviewed to public status.
+	if isPublic {
+		p.RecordMetadata.Status = backend.StatusPublic
+		p.RecordMetadata.Iteration = 2
+		err = cmd.tstore.RecordSave(tstoreToken, p.RecordMetadata, p.Metadata,
+			p.Files)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// importAuthDetails saves the auth details blob into tstore for the provided
+// proposal token, if it exists.
+func importAuthDetails(auth *ticketvote.AuthDetails, tstoreToken []byte, cmd *importCmd) error {
+	// Save auth details blob, if it exists.
+	if auth != nil {
+		err := saveBlob(auth, dataDescriptorAuthDetails, tstoreToken, cmd)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// importVoteDetails saves the vote details blob into tstore for the provided
+// proposal token, if it exists.
+func importVoteDetails(vote *ticketvote.VoteDetails, tstoreToken []byte, cmd *importCmd) error {
+	// Save vote details blob, if it exists.
+	if vote != nil {
+		err := saveBlob(vote, dataDescriptorVoteDetails, tstoreToken, cmd)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func importJournals(prop *proposal, tstoreToken []byte, cmd *importCmd) error {
+	// Threads for importing comments blobs and votes blobs.
+	var wg sync.WaitGroup
+
+	// Spin thread for saving vote blobs.
+	wg.Add(1)
+	go func(p *proposal, token []byte, cmd *importCmd) {
+		defer wg.Done()
+		err := importVotes(p, token, cmd)
+		if err != nil {
+			e := fmt.Errorf("saveVotes thread err: %w", err)
+			panic(e)
+		}
+	}(prop, tstoreToken, cmd)
+
+	// Spin thread for saving comment blobs.
+	wg.Add(1)
+	go func(p *proposal, token []byte, cmd *importCmd) {
```

Comment:
I would have a single saveBlobsConcurrently method that is generic and is used anytime you're saving blobs concurrently.